LEAN-3751

mnatsheh opened the pull request:
In this PR I attempt to remove the "mathjax-full" dependency (which is a significant portion of our bundle size). Although it's possible to directly include a `<script>` tag to load the library, I decided to programatically inject it . This is mainly to keep the loading and the usage of the library in the same place and not split across different repos.

mbartenev-atypon commented on src/lib/math.ts:
```diff
@@ -0,0 +1,73 @@
+}
+
+export const initMathJax = () => {
+  // @ts-ignore
```
Is the mathjax lib loaded via <script> smaller than the code that we load into our bundle? Can we create a separate bundle for mathjax during our build process and load it here instead of loading it from cdn? My only concern is that we make it a bit of a hidden thing that we use mathjax and that we use that version of it, because now if you'll check package json it's not there, although our project depends on it. So generation of our local separate bundle for that maybe more work, but maybe a better approach in terms of dependencies management.

mnatsheh replied:
It's a bit smaller, yes. However, the main benefit of this way compared to the old way is that the mathjax script can be cached by the browser. 

It is possible to use NPM and (host our own copy)[https://www.npmjs.com/package/mathjax#hosting-your-own-copy-of-the-mathjax-components] if that's preferred, but I'll have to look into how this interacts with vite.

mbartenev-atypon commented on src/lib/math.ts:
```diff
@@ -0,0 +1,73 @@
+}
+
+export const renderMath = (node: HTMLElement) => {
+  //@ts-ignore
```
maybe we should initialise mathjax here when a node attempts to render it? So we won't load it if there are no formulas in the doc?

mnatsheh replied:
I didn't do it here because if a user inserts an equation in a document that doesn't have an equation before, mathjax would not be loaded and the equation might not render correctly while the script is loading. However, this might be an edge case that we don't need to worry about.

mnatsheh commented:
I updated the code with the following:
- Added a dependency on mathjax. The dependency is not used directly, only its version.
- Similar to what we did in transform, I added a new script that generates a "versions.ts" file with the package version exported as a const, and also included the version of the mathjax dependency.
- Updated math.ts to use the generated version when constructing the URL.

This is still hacky but satisfies the requirements of everything being contained within the body-editor and mathjax version being specified in package.json.

mbartenev-atypon commented:
This looks acceptable to me. Since indeed, we preserve dependencies control function of package.json by going this way.
